### PR TITLE
Additional logs zed upload

### PR DESCRIPTION
**Pull request by giggsoff**
To have better understanding of behavior inside zedUpload we can improve logs

### CODE REVIEW

**Comment by deitch** [comment]
Do we use logs anywhere in here? These are libraries; no guarantee that logs are setup or that they are running somewhere that can print out.

File: libs/zedUpload/awsutil/s3sync.go [hunk 1]
```diff
@@ -220,19 +231,30 @@ func getNeededParts(cWriterOptions *writerOptions, bname, bkey string, doneParts
...
 				writtenBytes:        currentPartSize,
+				maxPartSize:         S3PartSize,
 			},
 			bname: bname,
 			bkey:  bkey,
 			start: S3PartSize*i + currentPartSize,
 			size:  S3PartSize - currentPartSize,
 		}
 		if i == partsCount-1 {
+			part.cWriter.maxPartSize = size - (partsCount-1)*S3PartSize
```
Comment (eriknordmark) [inline libs/zedUpload/awsutil/s3sync.go:242]: Why do you need to update maxPartSize? Shouldn't it be fixed? Or is this the last part which needs a different size?

File: libs/zedUpload/awsutil/s3sync.go [hunk 2]
```diff
@@ -8,10 +8,13 @@ import (
 	"compress/gzip"
 	"fmt"
 	"io"
+	"log"
```
Comment (eriknordmark) [inline libs/zedUpload/awsutil/s3sync.go:11]: It makes more sense to have the caller pass in some logger object by defining an log(ger) interface with Info* and Error* functions. That way we can have downloader pass in its log variable which is setup with logrus to do the right thing.

File: libs/zedUpload/awsutil/s3sync.go [hunk 3]
```diff
@@ -263,6 +285,19 @@ func (s *S3ctx) DownloadFile(fname, bname, bkey string,
 		}
 	}
 
+	// if PartSize differ from saved clean doneParts
+	// we may hit this in case of S3PartSize change or from different type of datastore
+	if doneParts.PartSize != S3PartSize {
+		// log only if parts provided
+		if len(doneParts.Parts) > 0 {
+			log.Printf("DownloadFile: stored PartSize (%d) is different from expected (%d), assume DownloadedParts are broken",
```
Comment (eriknordmark) [inline libs/zedUpload/awsutil/s3sync.go:293]: Is there also some place in zedUpload/awsutil where we can log when there is a retry of a part?

File: libs/zedUpload/awsutil/s3sync.go (hunk 1 above)
Comment (giggsoff) [reply libs/zedUpload/awsutil/s3sync.go:242]: Correct, the last part has different max size in general.

File: libs/zedUpload/awsutil/s3sync.go (hunk 2 above)
Comment (giggsoff) [reply libs/zedUpload/awsutil/s3sync.go]: Modified

File: libs/zedUpload/awsutil/s3sync.go (hunk 3 above)
Comment (giggsoff) [reply libs/zedUpload/awsutil/s3sync.go]: I put it few lines below (after requesting of getNeededParts).

**Review by eriknordmark — COMMENTED** [review]
LGTM

File: libs/zedUpload/awsutil/s3sync.go (hunk 1 above)
Comment (eriknordmark) [reply libs/zedUpload/awsutil/s3sync.go:241]: Would be good to add a comment to that affect e.g., // Set maxPartSize for last part to the remaining size

**Review by eriknordmark — APPROVED** [review]
LGTM. One comment nit though

File: libs/zedUpload/awsutil/s3sync.go (hunk 1 above)
Comment (giggsoff) [reply libs/zedUpload/awsutil/s3sync.go]: Modified

**Review by eriknordmark — APPROVED** [review]
Re-run eden

**Pull request merged**